test/jdk/sun/security/provider/acvp/ML_DSA_Test.java

```diff
@@ -83,15 +79,44 @@ static void keyGenTest(JSONValue kat, Provider p) throws Exception {
         }
     }
 
+    static String h2h(String in) {
+        return switch (in) {
+            case "SHA2-512/224" -> "SHA-512/224";
+            case "SHA2-512/256" -> "SHA-512/256";
+            case "SHA2-224" -> "SHA-224";
+            case "SHA2-256" -> "SHA-256";
+            case "SHA2-384" -> "SHA-384";
+            case "SHA2-512" -> "SHA-512";
+            case "SHAKE-128" -> "SHAKE128";
+            case "SHAKE-256" -> "SHAKE256";
+            default -> in;
+        };
+    }
+
     static void sigGenTest(JSONValue kat, Provider p) throws Exception {
         var s = p == null
                 ? Signature.getInstance("ML-DSA")
                 : Signature.getInstance("ML-DSA", p);
         for (var t : kat.get("testGroups").asArray()) {
             var pname = t.get("parameterSet").asString();
-            var det = Boolean.parseBoolean(t.get("deterministic").asString());
             System.out.println(">> " + pname + " sign");
+            var det = Boolean.parseBoolean(t.get("deterministic").asString());
+            if (t.get("signatureInterface").asString().equals("internal")) {
+                ML_DSA_Impls.version = ML_DSA_Impls.Version.DRAFT;
+            } else {
+                ML_DSA_Impls.version = ML_DSA_Impls.Version.FINAL;
+            }
+            if (t.get("externalMu").asString().equals("true")) {
+                continue; // Not supported
+            }
             for (var c : t.get("tests").asArray()) {
+                var cstr = c.get("context");
+                var ctxt = cstr == null ? new byte[0] : toByteArray(cstr.asString());
+                var hashAlg = c.get("hashAlg").asString();
+                var preHash = hashAlg.equals("none") ? null : h2h(hashAlg);
+                if (preHash != null || ctxt.length != 0) {
```

Comment:
We don't need to parse `hashAlg`. The check above can be simply:
```
if (!hashAlg.equals("none") || ctxt.length != 0) {
```
Same with the verification side. Then we can remove the `h2h` method.